### DISCUSSION TITLE
Why does #text accept unnamed size or fill argument?

### DISCUSSION THREAD

**Discussion by Muzosh** [typst/typst#5397] [Q&A]
I'm simply curious as to why does the text function accept size (`#text(17pt)[sample text]`) or color (`#text(blue)[sample text]`).

I wasn't able to find ANY indication that these two named arguments could be used in nameless manner.

**Comment by Muzosh** [comment]
Does this apply for other functions as well? How can I find whether a named argument can be used namelessly for a general function?

**Comment by laurmaedje** [comment]
It's purely for convenience. There are a few more cases (e.g. `page.paper`), but not many.

**Comment by ronleeon** [comment]
Even if this is convienient, I still feel this syntax is strange
